src/drivers/imx/sai.c

```diff
@@ -92,8 +96,12 @@ static void sai_start(struct dai *dai, int direction)
 	chan_idx = BIT(0);
 	/* RX3 supports capture on imx8ulp */
 #ifdef CONFIG_IMX8ULP
+	fifo_playback = (dai_fifo(dai, DAI_DIR_PLAYBACK) - dai_base(dai) - REG_SAI_TDR0) >> 2;
+	fifo_capture = (dai_fifo(dai, DAI_DIR_CAPTURE) - dai_base(dai) - REG_SAI_RDR0) >> 2;
```

Comment:
any reason not to only calculate one value under the respective `if`-branch below? Looks like you only need one variable in fact